simd/src/Kokkos_SIMD_AVX512.hpp

```diff
@@ -854,11 +854,11 @@ class const_where_expression<simd_mask<double, simd_abi::avx512_fixed_size<8>>,
                               static_cast<__m512d>(m_value), 8);
   }
 
-  friend constexpr auto const& Impl::mask<double, abi_type>(
-      const_where_expression<mask_type, value_type> const& x);
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+  value_type get_value() const { return m_value; }
 
-  friend constexpr auto const& Impl::value<double, abi_type>(
-      const_where_expression<mask_type, value_type> const& x);
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+  mask_type get_mask() const { return m_mask; }
```

Comment:
I'm all for reverting these back into member functions. My attempts to make these friend functions to work on host and device have not been successful without vast amount of code duplication.

But based on the [previous comments on these member functions]( https://github.com/kokkos/kokkos/pull/5913#discussion_r1139176096), I think renaming these to be `impl_get_value` and `impl_get_mask` may be preferred.